docs/components/index.md

```diff
@@ -0,0 +1,315 @@
+---
+title: Blazor components
+author: guardrex
+description: Learn how to create and use Blazor components, the fundamental building blocks of Blazor apps provided by compiled Razor or C# files.
+manager: wpickett
+ms.author: riande
+ms.custom: mvc
+ms.date: 04/09/2018
+ms.prod: asp.net-core
+ms.technology: aspnet
+ms.topic: article
+uid: client-side/blazor/components/index
+---
+# Blazor components
+
+By [Luke Latham](https://github.com/guardrex)
+
+[!INCLUDE[](~/includes/blazor-preview-notice.md)]
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/client-side/blazor/components/common/samples/)
+
+To download the sample:
+
+1. Download and unzip the [Blazor.Docs repository zip file](https://codeload.github.com/aspnet/Blazor.Docs/zip/master).
+1. Use the URL in the [sample link](https://github.com/aspnet/Docs/tree/master/aspnetcore/client-side/blazor/components/common/samples/) to navigate to the sample folder.
+
+A Blazor *component* represents any feature or collection of features that can be represented by a [Razor file (\*.cshtml)](https://docs.microsoft.com/aspnet/core/mvc/views/razor) or a [C# file (\*.cs)](/dotnet/csharp/getting-started/) compiled into a C# class assembly. Components are typical features found in web app client UIs, such as a page, dialog, or form, along with its processing logic. A component can also consist entirely of programming logic without UI elements. Components are the fundamental building blocks of Blazor apps and can be nested, reused, and shared between projects.
+
+## Use of Razor
+
+Razor is a syntax for embedding C# code into HTML markup files. Files containing Razor generally have a *\*.cshtml* file extension. We recommend that you become familiar with Razor while working with Blazor. For more information, see the [Razor syntax reference](https://docs.microsoft.com/aspnet/core/mvc/views/razor). Note that not all of the features of Razor are available in Blazor at this time.
+
+The following pair of Blazor components demonstrates how Blazor takes advantage of features in Razor, C#, and HTML. Many of these features are described in more detail later in this topic.
+
+Index page component (*Index.cshtml*):
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/Index.cshtml?start=1&end=11)]
+
+When the index page loads, the `HeadingComponent` loads. The component:
+
+* Displays a heading on the page.
+* Presents a form that allows the user to change the heading text and control its font style.
+
+Heading component (*HeadingComponent.cshtml*):
```

Comment:
This is quite a big chunk of code to put in as the very first component that a reader sees. I wonder if it would be easier to grasp if it were simplified, e.g., just to a checkbox that toggles whether the text is italic. What do you think?